docs: vault docs & migrating istio docs

emil2k opened the pull request:
This PR...

## Changes

- Guide to working with Vault.
- Migrating Guide to working with Istio.

## Fixes

-

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

olensmar commented:
how does this relate to https://docs-v2.testkube.io/articles/vault ?

olensmar commented:
never mind..

olensmar commented:
thanks - should these two really be under the Integrations category (which is more about runtime integrations)!? It seems to me they are more related to operations - so under installation?

emil2k commented:
@olensmar The existing Vault doc was under integrations so I updated in place, and added Istio there as well.
By "runtime", do you mean configurations related to the workflows, not some sort of global configuration?

The Istio one is focused on the Helm configurations necessary, but essentially one of the main outputs is a global worfklow template. The Vault guide is focused on setting up a workflow template and using it, so it is closer to "runtime". Not sure, where to place these, but let me know what you think is best and I can move them there.

emil2k commented:
Maybe it is better to move both of these to "more guides"?

rangoo94 commented on docs/articles/vault.md:
```diff
@@ -1,43 +1,194 @@
+  pod:
+    # highlight-next-line
+    serviceAccountName: test-vault
+    labels:
+      app: workflow-vault
```
Is that required to make Vault working?

emil2k replied:
No, this was just for my own debugging.

emil2k replied:
Removed.

rangoo94 commented on docs/articles/vault.md:
```diff
@@ -1,43 +1,194 @@
+  use:
+    - name: 'vault-secret-injection'
+  # highlight-end
+  container:
+    image: alpine:latest
```
Is that required?

emil2k replied:
I guess I can specify this at the step level to be more clear here. Does that make more sense?

rangoo94 replied:
I'm just thinking if it's needed at all, the shell script should work on the basic image as well, unless `alpine` has something specific

emil2k replied:
Oh, ok - I did not try the default. Let me try.

emil2k replied:
It works, switched to default image.

rangoo94 commented on docs/articles/vault.md:
```diff
@@ -1,43 +1,194 @@
+spec:
+  pod:
+    serviceAccountName: test-vault
+    labels:
+      app: workflow-vault
```
Same as above

rangoo94 commented on docs/articles/vault.md:
```diff
@@ -1,43 +1,194 @@
+      config:
+        port: "8201"
+  # highlight-end
+  container:
+    image: alpine:latest
```
Same as above

rangoo94 commented on docs/articles/vault.md:
```diff
@@ -1,43 +1,194 @@
+      traffic.sidecar.istio.io/excludeOutboundPorts: "{{ config.port }}"
+  after:
+  - name: 'Send quit signal to Vault agent'
+    container:
+      image: busybox
```
Is that required?

emil2k replied:
I think this one is required to make the `wget` request to quit the agent.

rangoo94 replied:
We do have `wget` and other Busybox binaries available for all the images automatically

rangoo94 reviewed (approved):
Some lines are I think not required, and probably better to remove them, so User won't think that i.e. they need `pod.labels` or some specific image. Other than that, LGTM 👍

jfermi commented on docs/articles/istio.md:
```diff
@@ -0,0 +1,120 @@
+configurations below to utilize Testkube until they can upgrade their cluster to
+enable native sidecars:
+
+### Configurations for Istio installations without native sidecars
```
```suggestion
### Configurations for Istio Installations Without Native Sidecars
```

jfermi commented on docs/articles/istio.md:
```diff
@@ -0,0 +1,120 @@
+
+All the values mentioned will be from the top of the specified chart.
+
+#### Disable Istio for test executions of prebuilt/container executors
```
```suggestion
#### Disable Istio for Test Executions of Prebuilt/Container Executors
```

jfermi commented on docs/articles/istio.md:
````diff
@@ -0,0 +1,120 @@
+        sidecar.istio.io/inject: "false"
+```
+
+#### Disable Istio for agent hooks
````
```suggestion
#### Disable Istio for Agent Hooks
```

jfermi commented on docs/articles/istio.md:
````diff
@@ -0,0 +1,120 @@
+        sidecar.istio.io/inject: "false"
+```
+
+#### Disable Istio for operator hooks
````
```suggestion
#### Disable Istio for Operator Hooks
```

jfermi commented on docs/articles/istio.md:
````diff
@@ -0,0 +1,120 @@
+        proxy.istio.io/config: '{ "holdApplicationUntilProxyStarts": true }'
+```
+
+#### Hold worker service till Istio's proxy is ready
````
```suggestion
#### Hold the Worker Service Until Istio's Proxy Is Ready
```

jfermi commented on docs/articles/vault.md:
```diff
@@ -1,43 +1,194 @@
 
-## Configurations for sidecar injector
+Prebuilt and container executors are deprecated so we do not officially support
+their usage with Vault, please migrate to workflows.
```
Should there be a link here to the docs to migrate to workflows?

emil2k replied:
Could you point me to the article, I am not able to find it.

olensmar replied:
https://docs-v2.testkube.io/articles/test-workflow-migration

jfermi commented on docs/articles/vault.md:
```diff
@@ -1,43 +1,194 @@
+:::
 
-Chart `testkube`:
+## Sidecar injector
```
```suggestion
## Sidecar Injector
```

jfermi commented on docs/articles/vault.md:
```diff
@@ -1,43 +1,194 @@
+sequence within init containers.
+- Configures the agent server running within the sidecar.
+- Calls the agent's quit endpoint after all steps of the workflow have
+  completed. Otherwise, the sidecar container would never exit keeping the
```
```suggestion
  completed. Otherwise, the sidecar container would never exit and the
```

jfermi commented on docs/articles/vault.md:
```diff
@@ -1,43 +1,194 @@
+- Configures the agent server running within the sidecar.
+- Calls the agent's quit endpoint after all steps of the workflow have
+  completed. Otherwise, the sidecar container would never exit keeping the
+workflow running indefinitely.
```
```suggestion
workflow would run indefinitely.
```

jfermi commented on docs/articles/vault.md:
```diff
@@ -1,43 +1,194 @@
+workflow running indefinitely.
+
+One can then reuse this workflow template in workflows requiring secret
+injection (example builds on Vault's great tutorial
```
```suggestion
injections (the example builds on Vault's great tutorial
```

jfermi commented on docs/articles/vault.md:
```diff
@@ -1,43 +1,194 @@
+
+### Troubleshooting
+
+#### Port conflict
```
```suggestion
#### Port Conflict
```

jfermi commented on docs/articles/vault.md:
````diff
@@ -1,43 +1,194 @@
 ```
 
-For the other executors, you can set these annotations globally with:
+#### Istio compatibility
````
```suggestion
#### Istio Compatibility
```

jfermi commented on docs/articles/vault.md:
```diff
@@ -1,43 +1,194 @@
-For the other executors, you can set these annotations globally with:
+#### Istio compatibility
+
+In setups, utilizing a combination of Vault and Istio we recommend explicitly
```
```suggestion
In setups utilizing a combination of Vault and Istio, we recommend explicitly
```

jfermi reviewed (commented):
Some editing suggestions.
